File: examples_cpp/c2_ifelse2.cpp
// C++ code
#include <iostream>
using namespace std;

int main() {
    float guess = 0.;
    cout << "Input your guess for the magic number: \n";
    cin >> guess;

    int magic_number = 25;
    int very_close_range = 3;

    if (guess == magic_number) {
        std::cout << "Congratulations! You've guessed the magic number!" << std::endl;
    } else if (guess >= (magic_number - very_close_range) && guess <= (magic_number + very_close_range)) {
        std::cout << "So close! You're within " << very_close_range << " numbers of the magic number." << std::endl;
    } else if (guess < magic_number) {
        std::cout << "Too low! Try guessing a higher number." << std::endl;
    } else {
        std::cout << "Too high! Try guessing a lower number." << std::endl;
    }

    return 0;
}

File: examples_py/c2_ifelse2.py
# Python code
guess = int(input("Input your guess for the magic number: "))
magic_number = 25
lower_bound = 20
upper_bound = 30
very_close_range = 3

if guess == magic_number:
    print("Congratulations! You've guessed the magic number!")
elif guess >= (magic_number - very_close_range) and guess <= (magic_number + very_close_range):
    print("So close! You're within ", very_close_range, " numbers of the magic number.")
elif guess < magic_number:
    print("Too low! Try guessing a higher number.")
else:
    print("Too high! Try guessing a lower number.")
